python/graphstorm/model/hgt_encoder.py

```diff
@@ -259,10 +259,12 @@ def forward(self, g, h):
                             trans_out = trans_out * alpha + h[k] * (1-alpha)
                     else:                       # Nodes not really in destination side.
                         warnings.warn("Warning. Graph convolution returned empty "
-                          f"dictionary, for node with type: {str(k)}")
-                        # So add psudo self-loop with feature copy.
-                        trans_out = self.drop(self.a_linears[k](h[k]))
-                        trans_out = trans_out * alpha + h[k] * (1-alpha)
+                          f"dictionary for node with type: {str(k)}. Pleaes check your data \
+                            for no in-degree {str(k)} nodes.")
+                        # So add psudo self-loop for the destination nodes with its own feature.
+                        dst_h = h[k][:g.num_dst_nodes(k)]
```

Comment:
So the case is there are dst_nodes but there is no in edge for those dst nodes, right?
Can you add a unit test in test_nn_model.py?

Reply:
Add a unit test for this case.